src/mqt/bench/targets/devices/ionq.py

```diff
@@ -10,55 +10,394 @@
+    cx_gate.append(GPI2Gate(1 / 4), [0])
+    cx_gate.append(MSGate(0, 0, 0.25), [0, 1])
+    cx_gate.append(GPI2Gate(1 / 2), [0])
+    cx_gate.append(GPI2Gate(1 / 2), [1])
+    cx_gate.append(GPI2Gate(-1 / 4), [0])
+    SessionEquivalenceLibrary.add_equivalence(CXGate(), cx_gate)
+
+
+def cx_via_zz_equivalence() -> None:
+    """Add equivalence CX ≡ H-ZZ(pi/4)-H."""
+    q = QuantumRegister(2, "q")
+    cx_equiv = QuantumCircuit(q)
+    cx_equiv.h(1)
+    cx_equiv.append(ZZGate(0.25), [0, 1])  # pi/4
+    cx_equiv.h(1)
+
+    SessionEquivalenceLibrary.add_equivalence(CXGate(), cx_equiv)
+
+
+# Below are the rules needed for Aer simulator to simulate circuits containing IonQ native gates
+
+
+def gpi_gate_equivalence() -> None:
+    """Add GPI gate equivalence to the SessionEquivalenceLibrary."""
+    q = QuantumRegister(1, "q")
+    phi_param = Parameter("phi_param")
+    gpi_gate = QuantumCircuit(q)
+    gpi_gate.append(XGate(), [0])
+    gpi_gate.append(RZGate(4 * phi_param * np.pi), [0])
+    SessionEquivalenceLibrary.add_equivalence(GPIGate(phi_param), gpi_gate)
+
+
+def gpi2_gate_equivalence() -> None:
+    """Add GPI2 gate equivalence to the SessionEquivalenceLibrary."""
+    q = QuantumRegister(1, "q")
+    phi_param = Parameter("phi_param")
+    gpi2_gate = QuantumCircuit(q)
+    gpi2_gate.append(RZGate(-2 * phi_param * np.pi), [0])
+    gpi2_gate.append(RXGate(np.pi / 2), [0])
+    gpi2_gate.append(RZGate(2 * phi_param * np.pi), [0])
+    SessionEquivalenceLibrary.add_equivalence(GPI2Gate(phi_param), gpi2_gate)
+
+
+def ms_gate_equivalence() -> None:
+    """Add MS gate equivalence to the SessionEquivalenceLibrary."""
+    q = QuantumRegister(2, "q")
+    phi0_param = Parameter("phi0_param")
+    phi1_param = Parameter("phi1_param")
+    theta_param = Parameter("theta_param")
+    alpha_param = phi0_param + phi1_param
+    beta_param = phi0_param - phi1_param
+    ms_gate = QuantumCircuit(q)
+    ms_gate.append(CXGate(), [1, 0])
+    ms_gate.x(0)
+    ms_gate.append(
+        CU3Gate(
+            2 * theta_param * np.pi,
+            2 * alpha_param * np.pi - np.pi / 2,
+            np.pi / 2 - 2 * alpha_param * np.pi,
+        ),
+        [0, 1],
+    )
+    ms_gate.x(0)
+    ms_gate.append(
+        CU3Gate(
+            2 * theta_param * np.pi,
+            -2 * beta_param * np.pi - np.pi / 2,
+            np.pi / 2 + 2 * beta_param * np.pi,
+        ),
+        [0, 1],
+    )
+    ms_gate.append(CXGate(), [1, 0])
+    SessionEquivalenceLibrary.add_equivalence(MSGate(phi0_param, phi1_param, theta_param), ms_gate)
+
+
+def zz_gate_equivalence() -> None:
+    """Add ZZ gate equivalence to the SessionEquivalenceLibrary."""
+    q = QuantumRegister(2, "q")
+    theta_param = Parameter("theta_param")
+    zz_circuit = QuantumCircuit(q)
+    zz_circuit.cx(0, 1)
+    zz_circuit.rz(2 * np.pi * theta_param, 1)
+    zz_circuit.cx(0, 1)
+    SessionEquivalenceLibrary.add_equivalence(ZZGate(theta_param), zz_circuit)
```

Comment:
I am pretty sure these are not needed for our purposes and might actually be covered by the `_define` or the the `__array__` method.

Reply:
I removed these methods.